src/fr/scanmanga/src/eu/kanade/tachiyomi/extension/fr/scanmanga/ScanManga.kt

```diff
@@ -78,119 +86,87 @@ class ScanManga : ParsedHttpSource() {
+
+        val statutText = document.select("div.titres_souspart").firstOrNull { it.text().contains("Statut") }?.ownText()
+        manga.status = when {
+            statutText?.contains("En cours", ignoreCase = true) == true -> SManga.ONGOING
+            statutText?.contains("Terminé", ignoreCase = true) == true -> SManga.COMPLETED
+            else -> SManga.UNKNOWN
         }
 
-        val jsonObj = json.parseToJsonElement(jsonRaw).jsonObject
-
-        val mangaList = jsonObj.entries.map { entry ->
-            SManga.create().apply {
-                title = Parser.unescapeEntities(entry.key, false)
-                genre = entry.value.jsonArray[2].jsonPrimitive.content.let {
-                    when {
-                        it.contains("0") -> "Shōnen"
-                        it.contains("1") -> "Shōjo"
-                        it.contains("2") -> "Seinen"
-                        it.contains("3") -> "Josei"
-                        else -> null
-                    }
-                }
-                status = entry.value.jsonArray[3].jsonPrimitive.content.let {
-                    when {
-                        it.contains("0") -> SManga.ONGOING // En cours
-                        it.contains("1") -> SManga.ONGOING // En pause
-                        it.contains("2") -> SManga.COMPLETED // Terminé
-                        it.contains("3") -> SManga.COMPLETED // One shot
-                        else -> SManga.UNKNOWN
-                    }
-                }
-                url = "/" + entry.value.jsonArray[0].jsonPrimitive.content + "/" +
-                    entry.value.jsonArray[1].jsonPrimitive.content + ".html"
-            }
-        }
+        manga.thumbnail_url = document.select("div.full_img_serie img[itemprop=image]").attr("src")
 
-        return MangasPage(mangaList, hasNextPage = false)
+        return manga
     }
 
-    override fun searchMangaSelector() = throw UnsupportedOperationException()
+    // Chapters
+    override fun chapterListSelector() = "div.chapt_m"
 
-    // Details
-    override fun mangaDetailsParse(document: Document): SManga = SManga.create().apply {
-        title = document.select("h2[itemprop=\"name\"]").text()
-        author = document.select("li[itemprop=\"author\"] a").joinToString { it.text() }
-        description = document.select("p[itemprop=\"description\"]").text()
-        thumbnail_url = document.select(".contenu_fiche_technique .image_manga img").attr("src")
-    }
+    override fun chapterFromElement(element: Element): SChapter {
+        val linkEl = element.selectFirst("td.publimg span.i a") ?: throw Exception("Lien chapitre introuvable")
+        val titleEl = element.selectFirst("td.publititle")
 
-    // Chapters
-    override fun chapterListSelector() = throw UnsupportedOperationException()
+        val chapterName = linkEl.text().trim()
+        val extraTitle = titleEl?.text()?.trim().orEmpty()
```

Comment:
`.text()` is already trimmed according to Jsoup docs
```suggestion
        val chapterName = linkEl.text()
        val extraTitle = titleEl?.text()?.orEmpty()
```